ships/0037-build-output-timestamp.md

```diff
@@ -0,0 +1,142 @@
+- Allow a Build user to specify that Shipwright is not suppose to do anything with regards to the container image creation timestamp.
+
+### Non-Goals
+
+- Modification of timestamps of the source files.
+- No post-processing of container images that used strategy managed push, i.e. Paketo Buildpacks.
+- No magic build strategy manipulation of steps other than the `image-processing` step.
+
+### User Stories
+
+#### Story 1
+
+As a Shipwright user, I want to use a `Dockerfile` based build where the resulting container image creation timestamp is set to unix timestamp zero.
+
+#### Story 2
+
+As a Shipwright user, I want to use build a container image that has the image creation timestamp that matches the one of the Git commit that was used for the build.
+
+#### Story 3
+
+As a Shipwright user, I want write a build that results in a build run that does not touch the image timestamp at all.
+
+## Proposal
+
+### API Changes
+
+We will extend the Build resource by an additional (Unix epoch) timestamp entry in `spec.output`:
+
+```yaml
+spec:
+  output:
+    timestamp: Zero | SourceTimestamp | <timestamp> | (null)
+```
+
+The `timestamp` entry is an optional string value field with the default being `null`. It can have multiple values that lead to different behavior:
+
+- `Zero` refers to the Unix epoch start date of January, 1st 1970.
+- `SourceTimestamp` would lead to taking the provided source code meta data to define the timestamp. In case the source is a Git repository, the Git commit that is used for the build defines the source timestamp. In any other case, the newest creation time stamp of the source files is used. Note, the timestamp needs to be taken from the user provided source files since the build process might introduce new files as part of the build itself. The source step needs to be extended to introduce a reliable point and single spot in which the source timestamp is being obtained.
+- `<timestamp>` must be a parsable Unix epoch timestamp that lies in the past and will be then used as-is for the creation timestamp of the image, i.e. custom neutral timestamp. An empty string is considered not parsable and therefore results in an error.
+- Null (unset) results in no timestamp usage at all, which means that the respective build strategy timestamp is used.
+
+The same fields will be made available in a BuildRun's `spec.output` and `spec.buildSpec.output`.
+
+### Backend
+
+The timestamp processing will be performed as part of the [image-processing step that is introduced with Shipwright-managed push](0026-shipwright-managed-push.md#backend).
+
+It can only work with Shipwright managed push strategies and it will rely on the source step providing the necessary timestamp piece of information. If not provided with a timestamp, it will fail.
```

Comment:
on failures, pls provide the proposed `Status`(should be False), the `Reason` and the `Description`. This should be part of the API proposed changes.